icu4c/source/i18n/decNumber.cpp

```diff
@@ -176,6 +176,13 @@
 /*      **  -- raise to the power                                     */
 /* ------------------------------------------------------------------ */
 
+// Notes:
+// Some of the functions are not used inside ICU and cause warnning. We
+// mark them with a macro
+// #ifdef U_NOT_USED_INSIDE_ICU_AND_CAUSE_WARNING
+// so it will not be compiled. 
+// Project which want to use them could define U_NOT_USED_INSIDE_ICU_AND_CAUSE_WARNING
+
```

Comment:
I don't understand this, all of this is internal to ICU, it couldn't possibly be used from anywhere else.

If this is something that ICU can be expected to start using sometime in the future, I'd say that it'd be much better to fix it now, instead of hiding it behind `#ifdef`s that never are defined which will increase the likelihood of bitrot.

But if this isn't anything that ICU is expected to ever use, why not simply delete it instead?

Reply:
@sffc explain this to me and I have hard time to explain it myself.

Reply:
ICU is the source of truth for the decNumber library, a popular library for decimals in C++.

Citation: https://speleotrove.com/decimal/decnumber.html

(also see Google Chat)

Reply:
ok